crates/rnote-engine/src/fileformats/rnoteformat/mod.rs

```diff
@@ -23,18 +23,31 @@ use std::io::{Read, Write};
 
 /// Compress bytes with gzip.
 fn compress_to_gzip(to_compress: &[u8]) -> Result<Vec<u8>, anyhow::Error> {
-    let mut encoder =
-        flate2::write::GzEncoder::new(Vec::<u8>::new(), flate2::Compression::default());
+    let mut encoder = flate2::write::GzEncoder::new(Vec::<u8>::new(), flate2::Compression::new(5));
     encoder.write_all(to_compress)?;
     Ok(encoder.finish()?)
 }
 
 /// Decompress from gzip.
 fn decompress_from_gzip(compressed: &[u8]) -> Result<Vec<u8>, anyhow::Error> {
+    // Optimization for the gzip format, defined by RFC 1952
+    // capacity of the vector defined by the size of the uncompressed data
+    // given in little endian format, by the last 4 bytes of "compressed"
+    //
+    //   ISIZE (Input SIZE)
+    //     This contains the size of the original (uncompressed) input data modulo 2^32.
+    let mut bytes: Vec<u8> = std::panic::catch_unwind(|| {
```

Comment:
TIL about that, very cool. But can this be refactored to not use catch unwind? I think it's not that great to use as a general try-catch mechanism, it should be possible to use fallible methods instead

Reply:
You're right, for some reason I didn't want this part of the code to fail for any reason, despite the fact that an error would be returned a couple of lines later if the file format was invalid, refactored :+1: